lib/ansible/modules/storage/netapp/na_ontap_ldap.py

```diff
@@ -0,0 +1,226 @@
+            attributes_list = result.get_child_by_name('attributes-list')
+            config_info = attributes_list.get_child_by_name('ldap-config')
+
+            # Define config details structure
+            config_details = {'client_config': config_info.get_child_content('client-config'),
+                              'skip_config_validation': config_info.get_child_content('skip-config-validation'),
+                              'vserver': config_info.get_child_content('vserver')}
+
+        return config_details
+
+    def create_ldap(self):
+        '''
+        Create LDAP configuration
+        '''
+        options = {
+            'client-config': self.parameters['name'],
+            'client-enabled': 'true'
+        }
+
+        if self.parameters.get('skip_config_validation') is not None:
+            options['skip-config-validation'] = self.parameters['skip_config_validation']
+
+        # Initialize NaElement
+        ldap_create = netapp_utils.zapi.NaElement.create_node_with_children('ldap-config-create', **options)
+
+        # Try to create LDAP configuration
+        try:
+            self.server.invoke_successfully(ldap_create, enable_tunneling=True)
+        except netapp_utils.zapi.NaApiError as errcatch:
+            self.module.fail_json(msg='Error creating LDAP configuration %s: %s' % (self.parameters['name'], to_native(errcatch)),
+                                  exception=traceback.format_exc())
+
+    def delete_ldap(self):
+        '''
+        Delete LDAP configuration
+        '''
+        ldap_client_delete = netapp_utils.zapi.NaElement.create_node_with_children('ldap-config-delete', **{})
+
+        try:
+            self.server.invoke_successfully(ldap_client_delete, enable_tunneling=True)
+        except netapp_utils.zapi.NaApiError as errcatch:
+            self.module.fail_json(msg='Error deleting LDAP configuration %s: %s' % (
+                self.parameters['name'], to_native(errcatch)), exception=traceback.format_exc())
+
+    def modify_ldap(self, modify):
+        '''
+        Modify LDAP
+        :param modify: list of modify attributes
+        '''
+        ldap_modify = netapp_utils.zapi.NaElement('ldap-config-modify')
+        ldap_modify.add_new_child('client-config', self.parameters['name'])
+
+        for attribute in modify:
+            if attribute == 'skip_config_validation':
+                ldap_modify.add_new_child('skip-config-validation', self.parameters[attribute])
+
+        # Try to modify LDAP
+        try:
+            self.server.invoke_successfully(ldap_modify, enable_tunneling=True)
+        except netapp_utils.zapi.NaApiError as errcatch:
+            self.module.fail_json(msg='Error modifying LDAP %s: %s' % (self.parameters['name'], to_native(errcatch)),
+                                  exception=traceback.format_exc())
+
+    def apply(self):
+        '''Call create/modify/delete operations.'''
+        current = self.get_ldap()
```

Comment:
Could you add at the top of Apply the following line. This will create an ems log event for users with auto support turned on.

netapp_utils.ems_log_event("na_ontap_ldap", self.server)

Reply:
Sure, resolved by commit 178b47169348d646fabf11b4b5915249dad6d6f7